### ISSUE TITLE
Improve handling of conflicting namespace imports

### ISSUE THREAD

**Issue by lukastaegert** [rollup/rollup#2944]
- Rollup Version: 1.15.5
- Operating System (or Browser): any
- Node Version: any

### How Do We Reproduce?
See this example:
https://rollupjs.org/repl?version=1.15.5&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QmxpYiU3RCUyMGZyb20lMjAnLiUyRmxpYi5qcyclM0IlNUNuY29uc29sZS5sb2cobGliKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmxpYi5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjAqJTIwZnJvbSUyMCdleHRlcm5hbDEnJTNCJTVDbmV4cG9ydCUyMColMjBmcm9tJTIwJ2V4dGVybmFsMiclM0IlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

When there is more than one namespace export, Rollup's handling could be improved. 

### Expected Behavior
Rollup checks both namespaces for the named import in some way.

### Actual Behavior
Currently, Rollup settles on assuming named exports are part of the first namespace reexport while it could be part of the second as well.

Related points: When there are internal namespace reexports with conflicting members, Rollup should at least warn, considering the actually correct behaviour would be to throw an error at runtime when a conflicting member is accessed.

**Comment by mischnic** [comment]
Another example (which doesn't seem covered by your examples with externals):

https://rollupjs.org/repl/?version=2.15.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBmcm9tJTIwJTVDJTIyLiUyRmEuanMlNUMlMjIlM0IlNUNuZXhwb3J0JTIwKiUyMGZyb20lMjAlNUMlMjIuJTJGYi5qcyU1QyUyMiUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmEuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwJTdCJTIwYyUyMCU3RCUyMGZyb20lMjAnLiUyRmMxLmpzJyUzQiU1Q25leHBvcnQlMjBjb25zdCUyMGElMjAlM0QlMjAlNUMlMjJhJTVDJTIyJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmIuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwJTdCJTIwYyUyMCU3RCUyMGZyb20lMjAnLiUyRmMyLmpzJyUzQiU1Q25leHBvcnQlMjBjb25zdCUyMGIlMjAlM0QlMjAlNUMlMjJiJTVDJTIyJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmMxLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGZ1bmN0aW9uJTIwYygpJTIwJTdCJTVDbiU1Q3RyZXR1cm4lMjBmYWxzZSUzQiU1Q24lN0QlMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyYzIuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZnVuY3Rpb24lMjBjKCklMjAlN0IlNUNuJTVDdHJldHVybiUyMHRydWUlM0IlNUNuJTdEJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

(This seems to be badly supported across the board - Babel's `plugin-transform-modules-commonjs` emits completely invalid output, it doesn't just use the wrong export.)

**Comment by guybedford** [comment]
@mischnic the example you provided is actually an error according to the spec, and you will get a special ambiguous export error in the browser or Node.js if running it natively.

**Comment by mischnic** [comment]
Yes, I know. But Rollup should probably still emit a warning (or throw an error) about these `conflicting namespace imports`

**Comment by guybedford** [comment]
Yes, Rollup should ideally make this a compilation error.

**Comment by guybedford** [comment]
This is a different feature to the error tracked in this issue though which is specifically for the externals cases. If you'd like to open a new issue to track the compilation error that would be great.
